wire-ios-request-strategy/Sources/Synchronization/Decoding/EventDecoder.swift

```diff
@@ -98,112 +115,117 @@ extension EventDecoder {
     /// - Parameters:
     ///   - encryptionKeys: Keys to be used to decrypt events.
     ///   - block: Event consume block which is called once for every stored event.
+
     public func processStoredEvents(with encryptionKeys: EncryptionKeys? = nil, _ block: ConsumeBlock) {
         process(with: encryptionKeys, block, firstCall: true)
     }
 
     /// Decrypts and stores the decrypted events as `StoreUpdateEvent` in the event database.
     /// The encryption context is only closed after the events have been stored, which ensures
     /// they can be decrypted again in case of a crash.
-    /// - parameter events The new events that should be decrypted and stored in the database.
-    /// - parameter startingAtIndex The startIndex to be used for the incrementing sortIndex of the stored events.
-    /// - Returns: Decrypted events
-    fileprivate func decryptAndStoreEvents(_ events: [ZMUpdateEvent], startingAtIndex startIndex: Int64) -> [ZMUpdateEvent] {
-        let account = Account(userName: "", userIdentifier: ZMUser.selfUser(in: self.syncMOC).remoteIdentifier)
-        let publicKey = try? EncryptionKeys.publicKey(for: account)
+    ///
+    /// - Parameters:
+    ///   - events The new events that should be decrypted and stored in the database.
+    ///   - startingAtIndex The startIndex to be used for the incrementing sortIndex of the stored events.
+    ///
+    /// - Returns: Decrypted events.
+
+    private func decryptAndStoreEvents(
+        _ events: [ZMUpdateEvent],
+        startingAtIndex startIndex: Int64
+    ) -> [ZMUpdateEvent] {
         var decryptedEvents: [ZMUpdateEvent] = []
 
-        // TODO: [John] use flag here
-        syncMOC.zm_cryptKeyStore.encryptionContext.perform { [weak self] (sessionsDirectory) -> Void in
+        // TODO: get core crypto lock
```

Comment:
We're going to need to provide a way to get the lock once and then decrypt several payloads without needing to acquire the lock again, but I'll deal with this in a separate PR.